export.py

```diff
@@ -0,0 +1,228 @@
+    name: Path
+
+    def __post_init__(self):
+        """
+        post-initialization processing and validation
+        """
+        self.checkpoint = Path(self.checkpoint)
+
+        if not self.checkpoint.exists():
+            raise FileNotFoundError(
+                f"The checkpoint {self.checkpoint} does " f"not exist."
+            )
+
+        self.image_shape = tuple(self.image_shape)
+        self.save_dir = Path(self.save_dir)
+        self.save_dir.mkdir(parents=True, exist_ok=True)
+        self.name = self.get_safe_name()
+
+    def get_safe_name(self):
+        if self.name:
+            self.name = Path(self.name)
+        else:
+            self.name = Path(self.checkpoint.with_suffix(".onnx").name)
+
+        filename = self.name.stem
+        self.name = self.save_dir / f"{filename}.onnx"
+        existence_counter = 0
+        while self.name.exists():
+            existence_counter += 1
+            self.name = self.name.parent / f"{filename}-{existence_counter}.onnx"
+        return self.name
+
+
+def parse_args() -> ExportArgs:
+    """
+    Add and parse arguments for exporting a yolact model to ONNX
+
+    :return: A ExportArgs Object with typed arguments
+    """
+    parser = argparse.ArgumentParser(description="Export yolact models to ONNX")
+
+    parser.add_argument(
+        "--checkpoint",
+        type=str,
+        required=True,
+        help="The yolact pytorch checkpoint to export",
+    )
+
+    parser.add_argument(
+        "--config",
+        "-c",
+        type=str,
+        default="yolact_base_config",
+        help="The config used to train the yolact model, for ex: "
+        "yolact_darknet53_config, yolact_resnet50_config, etc...; "
+        "Defaults to yolact_base_config.",
+    )
+
+    parser.add_argument(
+        "--recipe",
+        "-r",
+        type=str,
+        default=None,
+        help="Path or SparseZoo stub to the recipe used for training, "
+        "omit if no recipe used.",
+    )
+
+    parser.add_argument(
+        "--no-qat",
+        "-N",
+        action="store_true",
+        help="Flag to prevent conversion of a QAT(Quantization Aware Training) "
+             "Graph to a Quantized Graph",
+    )
+
+    parser.add_argument(
+        "--batch-size",
+        "-b",
+        type=int,
+        default=1,
+        help="The batch size to use while exporting the Model graph to ONNX;"
+        "Defaults to 1",
+    )
+
+    parser.add_argument(
+        "--image-shape",
+        "-S",
+        type=int,
+        nargs="+",
+        default=(3, 550, 550),
+        help="The image shape in (C, S, S) format to use for exporting the "
+        "Model graph to ONNX; Defaults to (3, 550, 550)",
+    )
+
+    parser.add_argument(
+        "--save-dir",
+        "-s",
+        type=str,
+        default="./exported_models",
+        help="The directory to save exported models to; "
+        'Defaults to "./exported_models"',
+    )
+
+    parser.add_argument(
+        "--name",
```

Comment:
let's combine save-dir and name into one argument and default the name to model.onnx or something if a directory is passed in